metricsaggregator/src/main/java/io/dockstore/metricsaggregator/MoneyStatistics.java

```diff
@@ -0,0 +1,90 @@
+package io.dockstore.metricsaggregator;
+
+import java.util.List;
+import org.javamoney.moneta.Money;
+
+/**
+ * Calculates money statistics in USD using the Java Money library to preserve accuracy.
+ */
+public class MoneyStatistics extends Statistics<Money> {
+    private static String currency = "USD";
+
+    private MoneyStatistics() {
+        super();
+    }
+
+    public MoneyStatistics(List<Money> dataPoints) {
+        super(dataPoints);
+    }
+
+    public MoneyStatistics(Money minimum, Money maximum, Money average, int numberOfDataPoints) {
+        super(minimum, maximum, average, numberOfDataPoints);
+    }
+
+    /**
+     * Create a new Statistics object from a list of statistics by aggregating the list of statistics
+     * @param statistics
+     * @return
+     */
+    public static MoneyStatistics createFromStatistics(List<MoneyStatistics> statistics) {
+        if (statistics.size() == 1) {
+            return statistics.get(0);
+        }
+
+        MoneyStatistics newStatistics = new MoneyStatistics();
+        newStatistics.setAverage(statistics);
+        newStatistics.setMinimum(statistics);
+        newStatistics.setMaximum(statistics);
+        newStatistics.setNumberOfDataPoints(statistics);
+        return newStatistics;
+    }
+
+    /**
+     * Get the lowest value from the list of data points.
+     * @param dataPoints
+     * @return
+     */
+    @Override
+    public Money calculateMinimum(List<Money> dataPoints) {
+        return dataPoints.stream()
+                .min(Money::compareTo)
+                .get();
+    }
+
+    /**
+     * Get the highest value from the list of data points.
+     * @param dataPoints
+     * @return
+     */
+    @Override
+    public Money calculateMaximum(List<Money> dataPoints) {
+        return dataPoints.stream()
+                .max(Money::compareTo)
+                .get();
+    }
+
+    /**
+     * Calculate the average from the list of data points.
+     * @param dataPoints
+     * @return
+     */
+    @Override
+    public Money calculateAverage(List<Money> dataPoints) {
+        Money sum = dataPoints.stream().reduce(Money.of(0, currency), Money::add);
+        return sum.divide(dataPoints.size());
+    }
+
+    /**
+     * Calculate a weighted average
+     */
+    @Override
+    public Money calculateWeightedAverage(List<? extends Statistics<Money>> statistics) {
+        int totalNumberOfDataPoints = getTotalNumberOfDataPoints(statistics);
+        return statistics.stream()
+                .map(stat -> {
+                    double weight = (double)stat.getNumberOfDataPoints() / (double)totalNumberOfDataPoints;
```

Comment:
Ok, nvm, the above. 
For my understanding, we still retain a "normal" `double` class for statistics where precision is less important and have a class using `Money` (which seems ok).

That said, I think averaging would be one operation which would be problematic (includes division and multiplying)

Reply:
> For my understanding, we still retain a "normal" double class for statistics where precision is less important and have a class using Money (which seems ok).
That said, I think averaging would be one operation which would be problematic (includes division and multiplying)

Should I change the [webservice](https://github.com/dockstore/dockstore/blob/8da78281c6881a4a543f1c2a84085867f5f272d7/dockstore-webservice/src/main/java/io/dockstore/webservice/core/metrics/StatisticMetric.java#L65) `StatisticMetric` fields to use `BigDecimal` instead of `double` then? This change would ripple to the metrics aggregator

Double checking, are you against the use of `double` to calculate the weight (on the line you commented on)? Should it be a `BigDecimal` instead?

Reply:
> That said, I think averaging would be one operation which would be problematic (includes division and multiplying)

This averaging is done with the Java Money class, i.e. `stat.getAverage()` is working with a `Money` object, not a double. The double is only used to calculate the weight.